lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -161,13 +188,47 @@ void HWMemSimImplPass::generateMemory(hw::HWModuleOp op, FirMemory mem) {
     wdata = addPipelineStages(b, numStages, clock, wdata);
 
     // Write logic
-    b.create<sv::AlwaysFFOp>(sv::EventControl::AtPosEdge, clock, [&]() {
-      auto wcond = b.createOrFold<comb::AndOp>(en, wmask);
-      b.create<sv::IfOp>(wcond, [&]() {
-        auto slot = b.create<sv::ArrayIndexInOutOp>(reg, addr);
-        b.create<sv::PAssignOp>(slot, wdata);
+    switch (mem.writeUnderWrite) {
+      // Undefined write order.  Lower each write port into a separate always
+      // block.
+    case 0:
```

Comment:
Plz use enums instead of magic number

Reply:
I moved the FIRRTL enum attr `WUWAttr` to the HW dialect and then use the associated enum here. Common logic in this section was also factored out into lambdas so its clearer what's going on in each case of the switch.